psst-gui/src/webapi/client.rs

```diff
@@ -213,8 +271,280 @@ impl WebApi {
+            #[serde(rename = "AUDIO")]
+            Audio,
+            #[serde(rename = "MIXED")]
+            Mixed,
+        }
+
+        #[derive(Deserialize)]
+        pub struct Publisher {
+            name: String,
+        }
+
+        #[derive(Deserialize)]
+        pub enum DataTypename {
+            Podcast,
+            Playlist,
+            Artist,
+            Album,
+        }
+
+        #[derive(Deserialize)]
+        pub struct Images {
+            items: Vec<ImagesItem>,
+        }
+
+        #[derive(Deserialize)]
+        #[serde(rename_all = "camelCase")]
+        pub struct ImagesItem {
+            sources: Vec<Source>,
+        }
+
+        #[derive(Deserialize)]
+        pub struct OwnerV2 {
+            data: OwnerV2Data,
+        }
+
+        #[derive(Deserialize)]
+        pub struct OwnerV2Data {
+            #[serde(rename = "__typename")]
+            name: String,
+        }
+
+        // Extract the playlists
+        let result: Welcome = self.load(request)?;
+        
+        let mut title: Arc<str> = Arc::from("");
+        let mut playlist: Vector<Playlist> = Vector::new();
+        let mut album: Vector<Arc<Album>> = Vector::new();
+        let mut artist: Vector<Artist> = Vector::new();
+        let mut show: Vector<Arc<Show>> = Vector::new();
+
+        result.data.home_sections.sections
+        .iter()
+        .for_each(|section| {
+            title = section.data.title.text.clone().into();
+                
+            section.section_items.items.iter().for_each(|item| {
+                let uri = item.content.data.uri.clone();
+                let id = uri.split(':').last().unwrap_or("").to_string();
+
+                match item.content.data.typename {
+                    DataTypename::Playlist => {
+                        playlist.push_back(Playlist {
+                            id: id.into(),
+                            name: Arc::from(item.content.data.name.clone().unwrap()),
+                            images: Some(item.content.data.images.as_ref().map_or_else(Vector::new, |images| 
+                                images.items.iter().map(|img| data::utils::Image {
+                                    url: Arc::from(img.sources.first().map(|s| s.url.as_str()).unwrap_or_default()),
+                                    width: None,
+                                    height: None,
+                                }).collect()
+                            )),
+                            description: {
+                                let desc = sanitize_str(&DEFAULT, item.content.data.description.as_deref().unwrap_or_default()).unwrap_or_default();
+                                // This is roughly 3 lines of description, truncated if too long
+                                if desc.chars().count() > 65 {
+                                    desc.chars().take(62).collect::<String>() + "..."
+                                } else {
+                                    desc
```

Comment:
There's gotta be a better way to do this, but I can't really see what it would be now.

Reply:
I think this is good for now, there is a contraint on the size of the size of the widget now so if it does end up onto the next line it doesn't really effect the view too much.